test/Microsoft.AspNetCore.Razor.Test/CodeGenerators/CSharpTagHelperRenderingTest.cs

```diff
@@ -21,6 +21,62 @@ public class CSharpTagHelperRenderingTest : TagHelperTestBase
         private static IEnumerable<TagHelperDescriptor> PrefixedPAndInputTagHelperDescriptors { get; }
             = BuildPAndInputTagHelperDescriptors(prefix: "THS");
 
+        private static IEnumerable<TagHelperDescriptor> CSSSelectorTagHelperDescriptors
```

Comment:
`[href$=]`, `[href^='']`, and `[href=""]` should be valid although only the last one will ever match.